protocol/xmrtaker/message_handler.go

```diff
@@ -296,31 +296,9 @@ func (s *swapState) handleNotifyXMRLock(msg *message.NotifyXMRLock) (net.Message
 		select {
 		case <-s.ctx.Done():
 			return
-		case <-time.After(until + time.Second):
-			s.Lock()
-			defer s.Unlock()
-
-			if !s.info.Status().IsOngoing() {
-				return
-			}
-
-			// XMRMaker hasn't claimed, and we're after t_1. let's call Refund
-			txhash, err := s.refund()
-			if err != nil {
-				log.Errorf("failed to refund: err=%s", err)
-				return
-			}
-
-			log.Infof("got our ETH back: tx hash=%s", txhash)
-			s.clearNextExpectedMessage(types.CompletedRefund) // TODO: duplicate?
-
-			// send NotifyRefund msg
-			if err = s.SendSwapMessage(&message.NotifyRefund{
-				TxHash: txhash.String(),
-			}, s.ID()); err != nil {
-				log.Errorf("failed to send refund message: err=%s", err)
-			}
-
+		// TODO: document why we add one second
+		case <-time.After(until + 5*time.Second):
+			s.handleT1Expired()
 			_ = s.Exit()
```

Comment:
This was the problem line right here. `Exit` grabs the lock, but we already have the lock on what was formerly line 300.

Reply:
@noot Based on other comments above, I can change this line from `s.Exit()` to `s.exit()` and move this line to the end `handleT1Expired()` below.

Reply:
Upon further inspection, `xmrtaker` only has an upper-case version of `Exit`. To make the change above, we would need to create a lower-case version (as is done in `xmrmaker`) that does not grab the lock.